crates/core/src/kernel/snapshot/mod.rs

```diff
@@ -763,6 +763,37 @@ mod datafusion {
     }
 }
 
+/// Retrieves a specific field from the schema based on the provided field name.
+/// It handles cases where the field name is nested or enclosed in backticks.
+fn get_stats_field<'a>(schema: &'a StructType, stats_field_name: &str) -> Option<&'a StructField> {
+    let dialect = sqlparser::dialect::GenericDialect {};
+    match sqlparser::parser::Parser::new(&dialect).try_with_sql(stats_field_name) {
+        Ok(mut parser) => match parser.parse_multipart_identifier() {
+            Ok(parts) => find_nested_field(schema, &parts, 0),
+            Err(_) => schema.field(stats_field_name),
+        }
+        Err(_) => schema.field(stats_field_name),
+    }
+}
+
+fn find_nested_field<'a>(schema: &'a StructType, parts: &[sqlparser::ast::Ident], part_idx: usize) -> Option<&'a StructField> {
+    let part_name = &parts[part_idx].value;
+    match schema.field(part_name) {
+        Some(field) => {
+            if part_idx == parts.len() - 1 {
+                Some(field)
+            } else {
+                match field.data_type() {
+                    DataType::Struct(struct_schema) => find_nested_field(struct_schema, parts, part_idx + 1),
+                    // Any part before the end must be a struct
+                    _ => None,
+                }
+            }
+        }
+        None => None,
+    }
+}
```

Comment:
This a little bit simpler?
```suggestion
fn find_nested_field<'a>(schema: &'a StructType, parts: &[sqlparser::ast::Ident]) -> Option<&'a StructField> {
    let part_name = &parts[0].value;
    match schema.field(part_name) {
        Some(field) => {
            if parts.len() == 1 {
                Some(field)
            } else {
                match field.data_type() {
                    DataType::Struct(struct_schema) => find_nested_field(struct_schema, parts[1..]),
                    // Any part before the end must be a struct
                    _ => None,
                }
            }
        }
        None => None,
    }
}
```

Reply:
nice! yes thank you

Reply:
Cool, if you can push that change, rest of code looks great! Also nice amount of tests 💪